ruby/ext/google/protobuf_c/defs.c

```diff
@@ -65,6 +65,10 @@ static upb_fielddef* check_field_notfrozen(const upb_fielddef* def) {
   return (upb_fielddef*)check_notfrozen((const upb_def*)def);
```

Comment:
upb_downcast_fielddef() / UPB_UPCAST(). Should be safer than doing casts directly.